src/multi-locale.ts

```diff
@@ -235,40 +285,44 @@ export function generateLocalizedAssets(
 		}));
 	};
 
-	// Apply after minification since we don't want to
-	// duplicate the costs of that for each asset
+	// When we have a custom localisation compiler,
+	// we have to run asset generation before minification,
+	// since the compiler may generate code which needs minifying.
+	// Otherwise, we can run after minification as an optimisation
 	if (isWebpack5Compilation(compilation)) {
 		const Webpack5Compilation = compilation.constructor as typeof WP5.Compilation;
-
+		const stage = localizeCompiler
+			? Webpack5Compilation.PROCESS_ASSETS_STAGE_DERIVED
```

Comment:
Problem with running post-minification is, the plugin’s first phase rewrites `__` calls into magic strings. So code like

```js
"foo" + __("bar")
```

gets rewritten into

```js
"foo" + "magic-base64-string"
```

which is liable to be minified into

```js
"foomagic-base64-string"
```

- which then can’t (correctly) be rewritten by the `localizeCompiler` because it’s inside a string literal. ([See comment in `multi-locale.ts`.](https://github.com/privatenumber/webpack-localize-assets-plugin/pull/21/files#diff-320c876878eda987cc99cc74b37e4ad33063c2daa6d359d92d7b907a698440d7R288)). [This test](https://github.com/privatenumber/webpack-localize-assets-plugin/pull/21/files#diff-6a0a8bbfc6a49ad17e94f931f95c6123f91d3476c9b1dbd41bb45a511a43b36aR539) fails when running post-minification.

This is part of the reason I arrived at the design in #25/#26.

Reply:
Ah I see. Perhaps we can replace it with a function call to preserve it as the "prefix" and "suffix":

```
localizeAssetsPluginSomeHash("magic-base64-string")
```

Reply:
This is an interesting idea. I’ll hack something up and see where it ends up

Reply:
https://github.com/privatenumber/webpack-localize-assets-plugin/pull/21/commits/81703e5b16e3d82c61cab870502fc6e650239ca4